### PR TITLE
[Flex] Adding support for OpenTelemetry

### PR DESCRIPTION
**Pull request by RohitRanjanMS**
Cherry picking OTel related commits to Flex release branch.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by pragnagopa — APPROVED** [review]

File: release_notes.md [hunk 1]
```diff
@@ -3,6 +3,11 @@
 <!-- Please add your release notes in the following format:
 - My change description (#PR)
 -->
+- Update Python Worker Version to [4.28.1](https://github.com/Azure/azure-functions-python-worker/releases/tag/4.28.1)
 - DotnetIsolated worker artifact clean up (#9976)
```
Comment (vrdmr) [inline release_notes.md:7]: Can we add OTel support info in Release notes?

**Review by vrdmr — APPROVED** [review]

File: release_notes.md (hunk 1 above)
Comment (RohitRanjanMS) [reply release_notes.md:7]: Not now, we are still trying to figure out release announcement for OTel.

**Review by pragnagopa — APPROVED** [review]

**Review by mathewc — APPROVED** [review]

**Comment by lsl-2020** [comment]
Hi @RohitRanjanMS, I saw the PR of supporting OTel is also merged in in Microsoft.Azure.Functions.Worker and a new version has been released. I am wondering when can we actually use the OTel support in Azure Function (isolated-worker mode)? Not sure how or when to bump up the Azure host version to include the latest support.

**Comment by RohitRanjanMS** [comment]
Hi @lsl-2020 , I would recommend you wait, our host is not ready yet. We will make an announcement when the feature is available.

**Comment by lsl-2020** [comment]
Hi @RohitRanjanMS, thank you for the heads up. Could you please try giving an ETA of it, even a rough one?

**Comment by vmcbaptista** [comment]
@lsl-2020 I've created an issue (#10258) that in a way asks support for what you're asking. @RohitRanjanMS can you give us an ETA or more info about this?

**Comment by RohitRanjanMS** [comment]
The feature is available, but it's limited to Flex consumption plan as of now. We will eventually release this for other hosting platforms.

https://learn.microsoft.com/en-us/azure/azure-functions/opentelemetry-howto?tabs=app-insights&pivots=programming-language-csharp

**Pull request merged**